sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -221,15 +186,15 @@ public RequestOptions addQueryParam(String parameterName, String value) {
      * @throws IllegalStateException if this instance is obtained by calling {@link RequestOptions#none()}.
      */
     public RequestOptions addQueryParam(String parameterName, String value, boolean encoded) {
```

Comment:
Should we remove `addQueryParam` method too since we've removed addHeader and setHeader?